src/openid4vci/Proof/verifyProof.ts

```diff
@@ -61,27 +61,27 @@ async function verifyJwtProof(proof: JwtProof, session: AuthorizationServerState
 	console.log("Proof header = ", proofHeader)
 	console.log("Proof body = ", JSON.parse(new TextDecoder().decode(base64url.decode(proof.jwt.split('.')[1]))))
 
-	const proofPayload = proofBodySchema.parse(JSON.parse(new TextDecoder().decode(base64url.decode(proof.jwt.split('.')[1]))));
-
-	const holderDID: string | undefined = proofPayload.iss ?? proofPayload.did; // did of the Holder
+	const holderDID: string | undefined = proofHeader.kid; // did of the Holder
```

Comment:
Just to be clear: this DID will be unique per credential, not a stable identifier shared across all of the user's credentials. Is that as expected?

Reply:
Yes. It is not in our scope yet to define how we will validate the public key and verify that it is associated with a valid wallet instance.

This is part of the trust framework that will be adopted in a later stage, which will  incorporate the Wallet Trust Evidence(WTE) and Wallet Instance Attestation(WIA) concepts. At the moment we remain "trust framework"-agnostic.